stdlib/public/runtime/SymbolInfo.h

```diff
@@ -43,14 +43,74 @@ namespace swift {
 struct SymbolInfo {
 private:
 #if defined(_WIN32) && !defined(__CYGWIN__)
-  SYMBOL_INFO_PACKAGE _package;
+  const void *_symbolAddress;
+  const char *_symbolName;
+  const char *_moduleFileName;
+  const void *_moduleBaseAddress;
 #elif SWIFT_STDLIB_HAS_DLADDR
   Dl_info _info;
 #endif
 
+#if defined(_WIN32) && !defined(__CYGWIN__)
+  SymbolInfo(const void *symbolAddress,
+             const char *symbolName,
+             const char *moduleFileName,
+             const void *moduleBaseAddress)
+    : _symbolAddress(symbolAddress),
+      _symbolName(symbolName),
+      _moduleFileName(moduleFileName),
+      _moduleBaseAddress(moduleBaseAddress)
+  {}
+#endif
+
 public:
   SymbolInfo() {}
 
+#if defined(_WIN32) && !defined(__CYGWIN__)
+  SymbolInfo(const SymbolInfo &other) {
+    _symbolName = nullptr;
+    _moduleFileName = nullptr;
+    *this = other;
+  }
+  SymbolInfo(SymbolInfo &&other) {
+    *this = std::move(other);
+  }
+  ~SymbolInfo() {
+    if (_moduleFileName)
+      ::free((void *)_moduleFileName);
+    if (_symbolName)
+      ::free((void *)_symbolName);
```

Comment:
`free(nullptr)` is well defined, I'm not sure that there is value in the explicit check.

Reply:
I'm aware that `::free(nullptr)` is well-defined; I didn't have the tests originally, then added them to match the ones in the copy assignment operator. I think you're right that we should remove them in the destructor.

(Nobody pointed out that I hadn't initialised them in the `SymbolInfo()` constructor; _that_ is clearly a bug, so I'll fix that and make this change separately.)